File: The Core/C#/isSmooth.cs
bool isSmooth(int[] arr) {
    bool result = false;
    
    if(arr[0] == arr[arr.Length - 1]){
        if(arr.Length % 2 == 0){
            if(arr[arr.Length/2 - 1] + arr[arr.Length/2] == arr[0]){
                result = true;
            }
        } else {
            if(arr[(int)Math.Floor((decimal)arr.Length/2)] == arr[0]){
                result = true;
            }
        }
    }
    
    return result;
}

File: The Core/List Forest Edge/40. Is Smooth/isSmooth.py
def isSmooth(arr):
    result = False
    
    if(arr[0] == arr[len(arr) - 1]):
        if(len(arr) % 2 == 0):
            if(arr[int(len(arr) / 2 - 1)] + arr[int(len(arr) / 2)] == arr[0]):
                result = True
        else:
            if(arr[int(math.floor(len(arr)) / 2)] == arr[0]):
                result = True
                
    return result
